pages/implementation/devs-101.mdx

```diff
@@ -0,0 +1,286 @@
+    case beer = "beer"
+    case wine = "wine"
+    case cocktail = "cocktail"
+    case water = "water"
+}
+```
+
+The second parameter has no constraints, so it is a `String` (though it is optional, because it's marked as optional in the tracking plan).
+
+In general, constraints can be added to properties to validate against:
+
+string properies: a finite list of allowed string values
+int and float properties: a minimum and/ maximum value
+object properties: exact structure of the allowed keys and types of values in the object
+
+> Avo can add helper parameters that are not defined in the tracking plan, but are known to be required to make that specific call, for example in server environments Avo usually adds a userId parameter to each function, so server can provide user id to the analytics tracking services.
+
+#### What's happening inside the Avo Functions
+
+Avo makes sure that the provided properties are correct according to the tracking plan with runtime validations, development invocations are reported to Avo servers so you can check the invocation status on the Avo website.
+If Avo Inspector is connected, the event shape is sent there. Read more about <Link passHref href="/implementation/devs-101#avo-inspector"><a>Avo Inspector</a></Link> and how to <Link passHref href="/implementation/start-using-inspector-with-avo-functions"><a>connect it to the Avo generated code</a></Link>.
+The last step is actually sending the provided data to all the configured tracking destinations. There are different classes of events that can be tracked: identify, log event, update user properties, unidentify, log revenue, log page view, but everything is handled inside the Avo file and delivered in a correct form to your destinations. If you are using a Custom Destination the data will be provided to the callback object you supplied during initialization.
+Tracking code would look similar to this:
+
+```swift
+// destination Amplitude
+var amplitudeDestinationEventProperties: [String: Any] = [:]
+// amplitudeDestinationEventProperties will be populated with all the provided properies to this Avo Function and all the system properties.
+amplitudeDestination?.logEvent("Test Unidentify", amplitudeDestinationEventProperties)
+amplitudeDestination?.unidentify()
+// destination Mixpanel
+var mixpanelDestinationEventProperties: [String: Any] = [:]
+// mixpanelDestinationEventProperties will be populated with all the provided properies to this Avo Function and all the system properties.
+mixpanelDestination?.logEvent("Test Unidentify", mixpanelDestinationEventProperties)
+mixpanelDestination?.unidentify()
+// destination Custom dest
+var customDestEventProperties: [String: Any] = [:]
+// customDestEventProperties will be populated with all the provided properies to this Avo Function and all the system properties.
+customDest.logEvent(eventName: "Test Unidentify", eventProperties: customDestEventProperties)
+customDest.unidentify()
+```
```

Comment:
```suggestion
```swift
// destination Amplitude
var amplitudeDestinationEventProperties: [String: Any] = [:]
// amplitudeDestinationEventProperties will be populated with all the provided properties to this Avo Function and all the system properties.
amplitudeDestination?.logEvent("Logged Out", amplitudeDestinationEventProperties)
amplitudeDestination?.unidentify()
// destination Mixpanel
var mixpanelDestinationEventProperties: [String: Any] = [:]
// mixpanelDestinationEventProperties will be populated with all the provided properties to this Avo Function and all the system properties.
mixpanelDestination?.logEvent("Logged Out", mixpanelDestinationEventProperties)
mixpanelDestination?.unidentify()
// destination Custom dest
var customDestEventProperties: [String: Any] = [:]
// customDestEventProperties will be populated with all the provided properies to this Avo Function and all the system properties.
customDest.logEvent(eventName: "Logged Out", eventProperties: customDestEventProperties)
customDest.unidentify()
```
```